components/ui/card-collection/card-collection.tsx

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from 'react';
+
+import { Card } from "../card";
+import { CardConfiguration, CardInputData, isFixedCardConfiguration } from "./card-collection-types";
+import { useCardCollectionConfiguration } from "./use-card-collection-configuration";
+
+interface CardCollectionProps {
+  cardInputData: CardInputData[];
+  columnCountToFill: number;
+}
+
+interface CardContentProps {
+  cardWidthClass: string;
+  cardConfiguration: CardConfiguration;
+  cardConfigurations: CardConfiguration[];
+}
+
+const CardContent = ({cardWidthClass, cardConfiguration, cardConfigurations}: CardContentProps) => {
+  const [previousCardWidthClass, setPreviousCardWidthClass] = useState<string>(cardWidthClass);
+
+  useEffect(() => {
+    if(cardWidthClass !== previousCardWidthClass) {
+      setPreviousCardWidthClass(cardWidthClass)
+
+      if(isFixedCardConfiguration(cardConfiguration)) {
+        return;
+      }
+
+      cardConfiguration.onCardSizeChange?.();
+    }
+  }, [previousCardWidthClass, cardWidthClass]);
```

Comment:
This `useEffect` and `useState` is why I needed to isolate `CardContent` into it's own thing. If I didn't need to do that, I would agree with you to just throw the render function into `CardCollection` and call it a day.